### ISSUE TITLE
poetry: broken on master

### ISSUE THREAD

**Issue by kierdavis** [NixOS/nixpkgs#57398]
## Issue description

`pythonPackages.poetry` fails to build since it cannot find a package satisfying `requests-toolbelt<0.9.0,>=0.8.0`. The version of `requests-toolbelt` currently provided in nixpkgs is `0.9.1`.

Cross referencing #55757 which contains the commit b86de1abec8e44772ba9630d87ceb8f9706cc73d which was probably the cause of this breakage.

### Steps to reproduce

```
[kier@coloris:~/checkouts/nixpkgs]$ git pull NixOS master
From github.com:NixOS/nixpkgs
 * branch                    master     -> FETCH_HEAD
Already up to date.

[kier@coloris:~/checkouts/nixpkgs]$ nix-build -A python3Packages.poetry
these derivations will be built:
  /nix/store/lw70s6f11n56bbakpkdwzqg5y7wdjpiq-python3.7-poetry-0.12.10.drv
building '/nix/store/lw70s6f11n56bbakpkdwzqg5y7wdjpiq-python3.7-poetry-0.12.10.drv'...
unpacking sources
unpacking source archive /nix/store/g8pkjkl01m87bxklp35iwg0wnliby0bz-poetry-0.12.10.tar.gz
source root is poetry-0.12.10
<snip>
adding 'poetry-0.12.10.dist-info/top_level.txt'
adding 'poetry-0.12.10.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
installing
/build/poetry-0.12.10/dist /build/poetry-0.12.10
Processing ./poetry-0.12.10-py3-none-any.whl
Requirement already satisfied: cachecontrol[filecache]<0.13.0,>=0.12.4 in /nix/store/paz52xm8whyal9vklzys9ak2d3mv8ziv-python3.7-CacheControl-0.12.5/lib/python3.7/site-packages (from poetry==0.12.10) (0.12.5)
Requirement already satisfied: tomlkit<0.6.0,>=0.5.1 in /nix/store/1cgqx3lzd749lr43cx31sy1pdm606zh3-python3.7-tomlkit-0.5.3/lib/python3.7/site-packages (from poetry==0.12.10) (0.5.3)
Requirement already satisfied: cleo<0.7.0,>=0.6.7 in /nix/store/p77fayqkfkkfllgly0dj79lkfqynwrgv-python3.7-cleo-0.7.2/lib/python3.7/site-packages (from poetry==0.12.10) (0.6.8)
Requirement already satisfied: cachy<0.3,>=0.2 in /nix/store/yr0jnxwrlx8pcndxynh5g6hqjkiaiqi5-python3.7-cachy-0.2.0/lib/python3.7/site-packages (from poetry==0.12.10) (0.2.0)
Requirement already satisfied: html5lib<2.0,>=1.0 in /nix/store/hfhg862bl9jgx946ywcg361jl2l1gp1b-python3.7-html5lib-1.0.1/lib/python3.7/site-packages (from poetry==0.12.10) (1.0.1)
Requirement already satisfied: shellingham<2.0,>=1.1 in /nix/store/386iyhs7rj5qlv981lfv72sqz0m083xc-python3.7-shellingham-1.2.8/lib/python3.7/site-packages (from poetry==0.12.10) (1.2.8)
Requirement already satisfied: pyparsing<3.0,>=2.2 in /nix/store/3hrcvnbsar9qalacvp9xr08bkz9hfxs4-python3.7-pyparsing-2.3.1/lib/python3.7/site-packages (from poetry==0.12.10) (2.3.1)
Requirement already satisfied: jsonschema<4.0,>=3.0.0a3 in /nix/store/2rnkpfpyd88nhnccr3s08rb9w8dpxy92-python3.7-jsonschema-3.0.0a3/lib/python3.7/site-packages (from poetry==0.12.10) (3.0.0a3)
Requirement already satisfied: pyrsistent<0.15.0,>=0.14.2 in /nix/store/5an6j4hybn8pvamva8mdc577v1237y01-python3.7-pyrsistent-0.14.11/lib/python3.7/site-packages (from poetry==0.12.10) (0.14.11)
Requirement already satisfied: pkginfo<2.0,>=1.4 in /nix/store/ci779zkwy8dz9snbd2zgjdrqkh50s4nq-python3.7-pkginfo-1.5.0.1/lib/python3.7/site-packages (from poetry==0.12.10) (1.5.0.1)
Collecting requests-toolbelt<0.9.0,>=0.8.0 (from poetry==0.12.10)
  Could not find a version that satisfies the requirement requests-toolbelt<0.9.0,>=0.8.0 (from poetry==0.12.10) (from versions: )
No matching distribution found for requests-toolbelt<0.9.0,>=0.8.0 (from poetry==0.12.10)
builder for '/nix/store/lw70s6f11n56bbakpkdwzqg5y7wdjpiq-python3.7-poetry-0.12.10.drv' failed with exit code 1
error: build of '/nix/store/lw70s6f11n56bbakpkdwzqg5y7wdjpiq-python3.7-poetry-0.12.10.drv' failed
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.105, NixOS, 18.09.2327.37694c8cc0e (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(root): `"nixos-18.09.2327.37694c8cc0e"`
 - channels(kier): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by ryanpeach** [comment]
Mine still says poetry is broken

**Comment by kierdavis** [comment]
@ryanpeach it was fixed in 19.03, but now it's apparently broken again in 19.09 (maybe for a different reason? I haven't had a chance to investigate).
